### PR TITLE
perl: add 5.26.2

### PR DESCRIPTION
**Pull request by davydden**
5.26.2 is the current stable version according to https://www.perl.org

### CODE REVIEW

File: var/spack/repos/builtin/packages/perl/package.py [hunk 1]
```diff
@@ -45,11 +45,13 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
     # URL must remain http:// so Spack can bootstrap curl
     url = "http://www.cpan.org/src/5.0/perl-5.24.1.tar.gz"
 
+    version('5.26.2', 'dc0fea097f3992a8cd53f8ac0810d523')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/perl/package.py]: According to http://www.cpan.org/src/README.html, this is a maintenance release. Can you put it under the "Maintenance releases" section? 

I know we have a policy of always sorting versions, but Perl apparently doesn't have that policy (even numbered releases are maintenance, odd numbered releases are development). Maybe you can add a link to this page and a comment explaining this.

File: var/spack/repos/builtin/packages/perl/package.py [hunk 2]
```diff
@@ -45,11 +45,13 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
...
+
     # Development releases
     version('5.25.11', '37a398682c36cd85992b34b5c1c25dc1')
 
     # Maintenance releases (recommended)
-    version('5.24.1', '765ef511b5b87a164e2531403ee16b3c', preferred=True)
+    version('5.24.1', '765ef511b5b87a164e2531403ee16b3c')
     version('5.22.3', 'aa4f236dc2fc6f88b871436b8d0fda95')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/perl/package.py]: Can you move 5.22.3 to the "Misc releases" section? Can you also add 5.22.4 to the "End of life" section?

Comment (davydden) [reply var/spack/repos/builtin/packages/perl/package.py]: 5.22.4? we don't have it in spack.

**Comment by davydden** [comment]
@adamjstewart do you want me to change something else in this perl PR?

**Comment by adamjstewart** [comment]
Can you add 5.22.4? I know it's not already in Spack, but people seem to need 5.22 for some reason and it contains all of the bug fixes on that branch. Might as well keep around the latest tip of each branch.

**Comment by davydden** [comment]
> Can you add 5.22.4?

sure, done.

File: var/spack/repos/builtin/packages/perl/package.py [hunk 3]
```diff
@@ -45,14 +45,19 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
...
 
-    # Maintenance releases (recommended)
-    version('5.24.1', '765ef511b5b87a164e2531403ee16b3c', preferred=True)
-    version('5.22.3', 'aa4f236dc2fc6f88b871436b8d0fda95')
+    # Maintenance releases (even numbers, recommended)
+    version('5.26.2', 'dc0fea097f3992a8cd53f8ac0810d523', preferred=True)
+    version('5.24.1', '765ef511b5b87a164e2531403ee16b3c')
 
     # Misc releases that people need
+    version('5.22.4', '31a71821682e02378fcdadeed85688b8')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/perl/package.py]: I would add it to "End of life releases". Honestly, maybe we should move all of these "Misc releases" to "End of life releases".

Comment (davydden) [reply var/spack/repos/builtin/packages/perl/package.py]: done.

**Pull request merged**